workers/unity/Assets/Playground/Scripts/MonoBehaviours/DeleteCubeCommandReceiver.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections.Generic;
+using Generated.Playground;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.Commands;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using Improbable.Worker.Core;
+using UnityEngine;
+
+#region Diagnostic control
+
+// Disable the "variable is never assigned" for injected fields.
+#pragma warning disable 649
+
+#endregion
+
+namespace Playground.MonoBehaviours
+{
+    public class DeleteCubeCommandReceiver : MonoBehaviour
+    {
+        [Require] private CubeSpawner.Requirables.CommandRequestHandler commandRequestHandler;
+        [Require] private CubeSpawner.Requirables.Writer writer;
+        [Require] private WorldCommands.Requirables.WorldCommandRequestSender worldCommandRequestSender;
+        [Require] private WorldCommands.Requirables.WorldCommandResponseHandler worldCommandResponseHandler;
+
+        private ILogDispatcher logDispatcher;
+        private const string CouldNotDeleteEntityWithId = "Could not delete entity with id {0}: {1}.";
+
+        private const string TheEntityHasBeenUnexpectedlyRemovedFromTheList =
+            "The entity {0} has been unexpectedly removed from the list.";
+
+        private readonly HashSet<long> sentRequestIds = new HashSet<long>();
+
+        public void OnEnable()
+        {
+            logDispatcher = GetComponent<SpatialOSComponent>().LogDispatcher;
+            commandRequestHandler.OnDeleteSpawnedCubeRequest += OnDeleteSpawnedCubeRequest;
+            worldCommandResponseHandler.OnDeleteEntityResponse += OnDeleteEntityResponse;
+        }
+
+        private void OnDeleteSpawnedCubeRequest(CubeSpawner.DeleteSpawnedCube.RequestResponder requestResponder)
```

Comment:
I distinctly remember this was meant to be removed

Reply:
I confused this with the pre-existing types when reading this. Which means it will also cause confusion for which you are meant to use elsewhere. Also I got very confused about the lack of something to send the response when everything else has one.

Reply:
Is that the `RequestResponder`? If that is the case it may be part of an open ticket

Reply:
What do you mean by "the lack of something to send the response", and by "everything else"?